src/test/shell/bazel/BUILD

```diff
@@ -134,18 +134,29 @@ sh_test(
 )
 
 sh_test(
-    name = "bazel_java_test",
+    name = "bazel_java_test_no_windows",
     size = "large",
     timeout = "eternal",
-    srcs = ["bazel_java_test.sh"],
-    data = [":test-deps"],
+    srcs = ["bazel_java_test_no_windows.sh"],
+    data = [
+        ":test-deps",
+    ],
     tags = [
-        # TODO(laszlocsomor): make this run on Windows. Currently fails because
-        # the java_common provider doesn't work on Windows.
         "no_windows",
     ],
 )
 
+sh_test(
+    name = "bazel_java_test",
+    size = "large",
+    timeout = "eternal",
```

Comment:
Are `large` and `eternal` necessary?

Reply:
I'm not sure yet, I was planning on investigating these tags in a subsequent PR. I'd keep the scope of this PR to enabling the tests on Windows.